### PR TITLE
Add check code quality errors for build

### PR DESCRIPTION
**Pull request by shalauniou**

### CODE REVIEW

File: src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy [hunk 1]
```diff
@@ -188,4 +193,37 @@ public class QualityAwareJavaPlugin implements Plugin<Project> {
         return targetFile.absolutePath
     }
 
+    private void createCheckCodeQualityErrorsTask(Project project) {
```
Comment (Andrei-Khadziukou) [inline src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: Could you please add brief description what it does in comments?

File: src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy [hunk 2]
```diff
@@ -188,4 +193,37 @@ public class QualityAwareJavaPlugin implements Plugin<Project> {
...
+        project.tasks.create(name: CHECK_CODE_QUALITY_ERRORS_TASK) {
+            description("Fail build if any Checkstyle, Pmd or Findbugs error found.")
+            doLast {
+                List reportErrors = []
+                project.sourceSets.each { sourceSet ->
+                    String sourceName = sourceSet.name
+                    List checkstyleReportErrors = CodeQualityUtil.getCheckstyleReportErrors(project, sourceName)
+                    if (!checkstyleReportErrors.isEmpty()) {
```
Comment (Andrei-Khadziukou) [inline src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: What happens if we don't perform such check?

Comment (shalauniou) [reply src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: by default `getCheckstyleReportErrors` retruns [], so we might have empty arrays  where we have no errors

Comment (Andrei-Khadziukou) [reply src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: I'm not sure it is a problem

Comment (shalauniou) [reply src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: It is easier to check for `empty`  or log it, if `checkstyle` or smth else has no errors - it shouldn't be added to reportErrors and checked one more time)

Comment (shalauniou) [reply src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: Updated

**Pull request merged**